bayesian/utility/security_vendor.py

```diff
@@ -0,0 +1,355 @@
+    : param version: The raw input version that needs to be converted.
+    : type return: semantic_version.base.Version
+    : return: The semantic version of raw input version.
+    """
+    conv_version = sv.Version.coerce('0.0.0')
+    try:
+        if version in ('', '-1', None):
+            version = '0.0.0'
+        """Needed for maven version like 1.5.2.RELEASE to be converted to
+        1.5.2 - RELEASE for semantic version to work."""
+        version = version.replace('.', '-', 3)
+        version = version.replace('-', '.', 2)
+        # Needed to add this so that -RELEASE is account as a Version.build
+        version = version.replace('-', '+', 3)
+        conv_version = sv.Version.coerce(version)
+    except ValueError:
+        logger.error(
+            f"Unexpected ValueError for the package {package_name} due to version {version}")
+        pass
+    finally:
+        return conv_version
+
+
+class VendorAnalyses:
+    """Vendor specific class for Component Analyses API v2."""
+
+    def __init__(self, ecosystem, package, version):
+        """For Flows related to Security Vendor Integrations."""
+        self.ecosystem = ecosystem
+        self.version = version
+        self.package = package
+        self.recommendation = dict()
+        self.response_data = dict()
+
+    @staticmethod
+    def is_package_known(query_result):
+        """Check if Package is Known or Unknown.
+
+        :return:
+            - True: Package info exists in GraphDb.
+            - False: Package info doesn't exists in Graphdb.
+        """
+        logger.info("Checking if package is known.")
+        if query_result is None:
+            return False
+        result_data = query_result.get('result', {}).get('data')
+        if not (result_data and len(result_data) > 0):
+            return False
+        return True
+
+    def get_vendor_analyses(self):
+        """Fetch analysis for given package+version from the graph database.
+
+        Fetch analysis for given package+version from the graph database.
+        Main Package for Building Vendor Response.
+        It finally builds and returns JSON Response. This is vendor specific Function.
+        :returns:
+            - Json Response
+            - None: On exceptions.
+        """
+        logger.info('Executing Vendor Specific Analyses')
+        try:
+            graph_response = GraphAnalyses.get_data_from_graph(
+                self.ecosystem, self.package, self.version, vendor='snyk')
+            if not self.is_package_known(graph_response):
+                # Trigger Unknown Flow on Unknown Packages
+                logger.info(f"Package {self.package} is not known.")
+                return None
+            return ResponseBuilder(
+                self.ecosystem, self.package, self.version).generate_recommendation(graph_response)
+
+        except Exception as e:
+            logger.error(str(e))
+            return None
+
+
+class ResponseBuilder():
```

Comment:
rename to componentanalysisresponsebuilder

Reply:
Ack!